pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -263,6 +266,122 @@ func (s *Server) loadRemoteCACerts(caOpts *caOptions, dir string) error {
 	return nil
 }
 
+// isValidCACertsFile As we are watching entire directory, but interested
+// in only 'ca-key.pem', 'ca-cert.pem', 'root-cert.pem' and 'cert-chain.pem'.
+// Events on other files are ignored.
+func isValidCACertsFile(path string) bool {
+	_, file := filepath.Split(path)
+
+	for _, name := range []string{"ca-cert.pem", "ca-key.pem", "root-cert.pem", "cert-chain.pem"} {
+		if file == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+// handleEvent handles the events on cacerts related files.
+// If create/write(modified) event occurs, then it verifies that
+// newly introduced cacerts are intermediate CA which is generated
+// from cuurent root-cert.pem. Then it updates and keycertbundle
+// and generates new dns certs.
+// TODO(rveerama1): Add support for new ROOT-CA rotation also.
+func handleEvent(s *Server) {
+	log.Info("Update Istiod cacerts")
+
+	currentCABundle := s.CA.GetCAKeyCertBundle().GetRootCertPem()
+	newCABundle, err := ioutil.ReadFile(path.Join(LocalCertDir.Get(), "root-cert.pem"))
+	if err != nil {
+		log.Error("failed reading root-cert.pem: ", err)
+		return
+	}
+
+	// Only updating intermediate CA is supported now
+	if !bytes.Equal(currentCABundle, newCABundle) {
+		log.Info("Updating new ROOT-CA not supported")
```

Comment:
log an error instead?

Reply:
log.err causing failures with tests. I am planning to add tests in another PR after this.

Reply:
SG, thanks!